fastlane/lib/fastlane/environment_printer.rb

```diff
@@ -132,10 +132,24 @@ def self.print_system_environment
       require "openssl"
 
       env_output = "### Stack\n\n"
-      product, version, build = `sw_vers`.strip.split("\n").map { |line| line.split(':').last.strip }
+      product, version, build = "Unkown"
+      os_version = "UNKOWN"
+
+      if FastlaneCore::Helper.mac?
+        product, version, build = `sw_vers`.strip.split("\n").map { |line| line.split(':').last.strip }
+        os_version = `sw_vers -productVersion`.strip
+      end
+
+      if FastlaneCore::Helper.linux?
```

Comment:
Linux doesn't have a single way to check for version information unfortunately, but some are described here:  http://unix.stackexchange.com/questions/88644/how-to-check-os-and-version-using-a-linux-command

It would be nice if we could fall back on multiple types of ways of checking the version information on Linux.